src/rauc-hawkbit-updater.c

```diff
@@ -118,6 +118,12 @@ int main(int argc, char **argv)
                 goto out;
         }
 
+        if (config_file == NULL) {
+                g_printerr("No configuration file given\n");
+                exit_code = 3;
```

Comment:
Would it be better to change the error_code to something different than 3? Three is already used below (No such configuration file).

Reply:
I also thought about that, but it was not clear to me what the error code says and I interpreted it as 'configuration file issue'. We could also return 4 here, but maybe we need a better definition of error return codes, anyway. I do not have a real strong opinion on this, yet. What would you prefer?

Reply:
I don't have a strong opinion. Imagine that we change it to 4. Could we then create an extra test case? I am thinking that we can have two different errors:
1. No configure file provided (null).
2. Config file provided but it does not exist.

Reply:
I've now changed the exit code and also added three more tests covering the cases a) invalid argument, b) missing config argument, c) path to non-existing config

Reply:
Great!